front/src/utils/duration.ts

```diff
@@ -9,20 +9,20 @@ export class Duration {
   /** Number of milliseconds */
   readonly ms: number;
 
-  constructor(ms: number) {
-    this.ms = ms;
+  constructor({ hours = 0, minutes = 0, seconds = 0, milliseconds = 0 }) {
+    this.ms = ((hours * 60 + minutes) * 60 + seconds) * 1000 + milliseconds;
   }
 
-  static zero = new Duration(0);
+  static zero = new Duration({});
 
   /** Parse an ISO 8601 duration string. */
```

Comment:
As the function does not only accept durations in ISO 8601 format
```suggestion
  /** Parse a duration string using dayjs. */
```

Reply:
This function is not changed in this PR.

The intent of the function is to only support ISO 8601 duration strings, and nothing else. I don't want to tie the `Duration` API to a particular library.

AFAIK since the parameter is a `string` dayjs will not accept other formats.